services/users-api/lambdas/deleteAttachment.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-console */
+import to from 'await-to-js';
+import { throwError, BadRequestError } from '@helsingborg-stad/npm-api-error-handling';
+import S3 from '../../../libs/S3';
+import * as response from '../../../libs/response';
+import { decodeToken } from '../../../libs/token';
+
+const BUCKET_NAME = process.env.BUCKET_NAME;
+
+export async function main(event) {
+  const { filename } = event.pathParameters;
+  if (!filename) {
+    return response.failure(new BadRequestError('Missing filename in path query string'));
+  }
+
+  const decodedToken = decodeToken(event);
+  const { personalNumber } = decodedToken;
+
+  const [getFilesError, userS3Files] = await to(getFilesFromUserS3Bucket(personalNumber));
+  if (getFilesError) {
+    return response.failure(getFilesError);
+  }
+
+  const fileToDelete = `${personalNumber}/${filename}`;
+
+  const [findFileError] = await to(findFile(userS3Files, fileToDelete));
+  if (findFileError) {
+    return response.failure(findFileError);
+  }
+
+  const [deleteFileError] = await to(deleteFile(fileToDelete));
+  if (deleteFileError) {
+    return response.failure(deleteFileError);
+  }
+
+  return response.success(200, {
+    type: 'userAttachment',
+  });
+}
+
+async function getFilesFromUserS3Bucket(personalNumber) {
+  const [getFilesError, s3Files] = await to(S3.getFiles(BUCKET_NAME, `${personalNumber}/`));
+  if (getFilesError) {
+    throwError(getFilesError.statusCode, getFilesError.message);
+  }
+
+  return s3Files;
+}
+
+async function deleteFile(key) {
```

Comment:
fileKey might be a better argument name, the key argument here raises question of what key we are expecting.

Reply:
Corrected with fileKey